Source/WebKit/UIProcess/_WKWebViewPrintFormatterInternal.h

```diff
@@ -33,16 +33,24 @@
 @end
 
 @interface _WKWebViewPrintFormatter ()
+- (BOOL)_shouldDrawUsingBitmap;
 - (void)_setSnapshotPaperRect:(CGRect)paperRect;
 - (void)_setPrintedDocument:(CGPDFDocumentRef)printedDocument;
+- (void)_setPrintPreviewImage:(CGImageRef)printPreviewImage;
+- (void)_invalidatePrintRenderingState;
 
-- (void)_waitForPrintedDocument;
+- (void)_waitForPrintedDocumentOrImage;
 @end
 
 @protocol _WKWebViewPrintProvider <NSObject>
+@property (nonatomic, readonly) BOOL _wk_printFormatterRequiresMainThread;
+
+- (BOOL)_wk_supportsRenderingBitmap;
```

Comment:
Why this extra method instead of just checking if the provider implements the optional method?